File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/disk/db/DBHelper.java
package com.hgianastasio.biblioulbrav2.data.disk.db;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.hgianastasio.biblioulbrav2.data.R;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

import javax.inject.Inject;

/**
 * Created by heitorgianastasio on 4/26/17.
 */

public class DBHelper extends SQLiteOpenHelper {
    private Context mContext;
    private static final String DB_NAME = "BiblioULBRADB";
    private static final int DB_VERSION = 1;

    @Inject
    public DBHelper(Context context) {
        super(context, DB_NAME, null, DB_VERSION);
        mContext = context;
    }

    private String fromRawResource(int resourceId){
        InputStream inputStream = mContext.getResources().openRawResource(resourceId);
        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
        String result = "";
        try {
            do{
                result = result.concat(reader.readLine());
            }while (reader.ready());

        } catch (IOException e) {
            e.printStackTrace();
        }
        return result;
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(fromRawResource(R.raw.sql_createtable_historybooks));
        db.execSQL(fromRawResource(R.raw.sql_createtable_loanbooks));
    }

    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {

    }


}

File: data/src/main/java/com/hgianastasio/biblioulbrav2/data/disk/db/DBHelper.kt
package com.hgianastasio.biblioulbrav2.data.disk.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import com.hgianastasio.biblioulbrav2.data.R
import java.io.BufferedReader
import java.io.IOException
import java.io.InputStreamReader
import javax.inject.Inject

/**
 * Created by heitorgianastasio on 4/26/17.
 */
class DBHelper @Inject constructor(private val mContext: Context) : SQLiteOpenHelper(mContext, DB_NAME, null, DB_VERSION) {
    private fun fromRawResource(resourceId: Int): String {
        val inputStream = mContext.resources.openRawResource(resourceId)
        val reader = BufferedReader(InputStreamReader(inputStream))
        var result = ""
        try {
            do {
                result += reader.readLine()
            } while (reader.ready())
        } catch (e: IOException) {
            e.printStackTrace()
        }
        return result
    }

    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(fromRawResource(R.raw.sql_createtable_historybooks))
        db.execSQL(fromRawResource(R.raw.sql_createtable_loanbooks))
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {}

    companion object {
        private const val DB_NAME = "BiblioULBRADB"
        private const val DB_VERSION = 1
    }

}